### DISCUSSION TITLE
[gatsby-source-contentful] Error: VipsJpeg: Premature end of JPEG file

### DISCUSSION THREAD

**Discussion by kimthanh932000** [gatsbyjs/gatsby#32200] [Help] [answered]
I'm using `...GatsbyContentfulFluid_tracedSVG` to apply preview image feature for images fetching from `contentful` and it works fine at my local. But I get `Error: VipsJpeg: Premature end of JPEG file` randomly at build time. I tried to resize the images but it still didn't work. Is there any way that I can fix it? Or is it a bug of the `plugin` isfelf?

**component**

```
import { graphql } from "gatsby";
import Img from "gatsby-image";

const BlogImage = ({ data }) => {
       return (
            <Img fluid={data.featureImage.fluid} />
       )
}
export const pageQuery = graphql`
            query {
                  featuredImage {
	                fluid {
		           ...GatsbyContentfulFluid_tracedSVG
	                }
            }
}`
```
**terminal**

![Screenshot from 2021-07-02 08-49-04](https://user-images.githubusercontent.com/30076762/124209209-51502300-db13-11eb-893f-2f99e2305e57.png)

**package.json**

```
"dependencies": {
		"@fontsource/lato": "^4.2.2",
		"@fontsource/montserrat": "^4.2.2",
		"axios": "^0.21.1",
		"focus-trap-react": "^8.5.1",
		"gatsby": "^2.23.9",
		"gatsby-image": "^3.8.0",
		"gatsby-plugin-react-helmet": "^3.3.5",
		"gatsby-plugin-sharp": "^2.6.43",
		"gatsby-plugin-web-font-loader": "^1.0.4",
		"gatsby-source-contentful": "^2.3.17",
		"gatsby-source-filesystem": "^3.3.0",
		"gatsby-source-twitter": "^4.0.0",
		"gatsby-transformer-remark": "^2.8.18",
		"gatsby-transformer-sharp": "^2.5.6",
		"gh-pages": "^3.1.0",
		"joi-browser": "13.4",
		"lodash": "^4.17.15",
		"prop-types": "^15.7.2",
		"react": "^16.13.1",
		"react-dom": "^16.13.1",
		"react-google-recaptcha-v3": "^1.9.4",
		"react-helmet": "^6.1.0",
		"react-share": "^4.4.0",
		"react-slick": "^0.28.1",
		"react-textarea-autosize": "^8.3.2",
		"react-twitter-embed": "^3.0.3",
		"slick-carousel": "^1.8.1"
	}


```

**Comment by axe312ger** [comment, marked as answer]
Rarely I can see similar behaviour in my CI builds. My guess is network issues, especially older versions of gatsby-source-contentful did request a lot of images at the same time which might lead to broken files.

I wonder if we should catch such errors and retry (once?) including re-downloading the file.

**Reply by kimthanh932000** [reply]
I installed the newer version of gatsby-source-contentful and now I don't get such error anymore. The version is ^4.6.3.

**Reply by axe312ger** [reply]
Perfect. That would mean my fixes helped you :)
